src/common/k8s-api/api-kube.injectable.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import assert from "assert";
+import { apiKubePrefix, isDevelopment } from "../vars";
+import { createStoresAndApisInjectionToken } from "./create-stores-apis.token";
+import { KubeJsonApi } from "./kube-json-api";
+
+const apiKubeInjectable = getInjectable({
+  id: "api-kube",
+  instantiate: (di) => {
+    assert(di.inject(createStoresAndApisInjectionToken), "apiKube is only available in certain environments");
```

Comment:
Nit: Hmm. I believe `di` should make it apparent that implementation for a token is not registered? Anyway the better error is appreciated, yet it could be better still, eg. what are these specific environments?

Reply:
It's also possible that the assert never gets to do anything, as `di` will throw error first about the inject of something unregistered.

Reply:
It is not that it is unregistered but that in renderer it is sometimes `false` (main frame) and sometimes `true` (cluster frame).

This also helps with testing as it is easy to just override this.

Reply:
> It is not that it is unregistered ...

Ah, I see.

Creating something and receiving data about something different are separate responsibilities -> violation of Single Responsibility Principle.

We could have both clarity, and easy overridability. Could there be a `areWeInRootFrameOrClusterFrameInjectable`, at least for a while?

Remember: the root cause of all this complexity is that there is the root `frame`, and also the cluster `iframes` which all run the same monolithic code, when ideally they should be considered as if separate applications with their own code. Or better yet, there should only be one application for a window of Lens.

Reply:
> `areWeInRootFrameOrClusterFrameInjectable`

That is basically what this is. And yes I know that this complexity is from the frames.

Reply:
Sure, but in addition to returning if `areWeInRootFrameOrClusterFrameInjectable`, current code also "creates stores and  API's", as an additional responsibility. Also the name `createStoresAndApisInjectionToken` is far from `areWeInRootFrameOrClusterFrameInjectable`.